# Section Summary: Statistical Analysis

## Goal

The objective of this notebook is to validate hypotheses about the Airbnb market using statistical tests and quantify the strength of relationships between key variables.

In [1]:
import pandas as pd
import scipy.stats as stats

In [2]:
# Load enriched dataset

df = pd.read_csv(
    "../data/processed/enriched_listings.csv"
)

In [4]:
# Section 5.1: Hypothesis Testing
# Task: Compare prices between entire homes and private rooms

entire_home_prices = df[
    df["room_type"] == "Entire home/apt"
]["price"]

private_room_prices = df[
    df["room_type"] == "Private room"
]["price"]

t_stat, p_value = stats.ttest_ind(
    entire_home_prices,
    private_room_prices,
    equal_var=False,
    nan_policy="omit"
)

print(f"T-statistic: {t_stat:.4f}")
print(f"P-value: {p_value:.10f}")

T-statistic: 20.4961
P-value: 0.0000000000


## Hypothesis 1 Results

### Hypotheses

- **Null hypothesis (H₀):** There is no significant difference in prices between entire homes and private rooms.
- **Alternative hypothesis (H₁):** Entire homes and private rooms have different prices.

### Results

- **T-statistic:** 20.4961
- **P-value:** < 0.001

### Conclusion

Since the p-value is significantly smaller than the significance level of 0.05, we reject the null hypothesis.

There is strong statistical evidence that room type has a significant effect on Airbnb listing prices. Entire homes and apartments tend to be more expensive than private rooms.

In [5]:
# Section 5.1: Hypothesis Testing
# Task: Compare ratings between superhosts and non-superhosts

superhost_ratings = df[
    df["host_is_superhost"] == "t"
]["review_scores_rating"]

non_superhost_ratings = df[
    df["host_is_superhost"] == "f"
]["review_scores_rating"]

t_stat, p_value = stats.ttest_ind(
    superhost_ratings,
    non_superhost_ratings,
    equal_var=False,
    nan_policy="omit"
)

print(f"T-statistic: {t_stat:.4f}")
print(f"P-value: {p_value:.10f}")

T-statistic: 35.6978
P-value: 0.0000000000


## Hypothesis 2 Results

### Hypotheses

- **Null hypothesis (H₀):** There is no significant difference in ratings between superhosts and non-superhosts.
- **Alternative hypothesis (H₁):** Superhosts and non-superhosts have different ratings.

### Results

- **T-statistic:** 35.6978
- **P-value:** < 0.001

### Conclusion

Since the p-value is significantly smaller than the significance level of 0.05, we reject the null hypothesis.

There is strong statistical evidence that superhosts receive different ratings compared to non-superhosts. This suggests that host status is associated with guest satisfaction and overall review scores.

In [6]:
# Section 5.1: Hypothesis Testing
# Task: Compare prices between superhosts and non-superhosts

superhost_prices = df[
    df["host_is_superhost"] == "t"
]["price"]

non_superhost_prices = df[
    df["host_is_superhost"] == "f"
]["price"]

t_stat, p_value = stats.ttest_ind(
    superhost_prices,
    non_superhost_prices,
    equal_var=False,
    nan_policy="omit"
)

print(f"T-statistic: {t_stat:.4f}")
print(f"P-value: {p_value:.10f}")

T-statistic: -2.0993
P-value: 0.0358006115


## Hypothesis 3 Results

### Hypotheses

- **Null hypothesis (H₀):** There is no significant difference in prices between superhosts and non-superhosts.
- **Alternative hypothesis (H₁):** Superhosts and non-superhosts have different prices.

### Results

- **T-statistic:** -2.0993
- **P-value:** 0.0358

### Conclusion

Since the p-value is smaller than the significance level of 0.05, we reject the null hypothesis.

There is statistical evidence that listing prices differ between superhosts and non-superhosts. However, the relatively small t-statistic suggests that the difference is less pronounced than the differences observed in the previous hypotheses.

### Business Interpretation

While superhosts receive significantly higher ratings, their prices are not substantially different from those of non-superhosts. This suggests that guests may value service quality, responsiveness, and reliability in addition to price.